File: csharp/grains/Grains.cs
using System;
using System.Linq;

public static class Grains
{
    public static double Square(int n)
    {
        if (n <= 0 || n > 64) throw new ArgumentOutOfRangeException();
        
        return Math.Pow(2, n-1);
    }

    public static double Total() =>
        Enumerable
            .Range(1, 64)
            .Select(Square)
            .Sum();
}

File: nim/grains/grains.nim
import std/math
  
proc onSquare*(n: int): uint64 =
  if n <= 0 or n > 64:
    raise newException(ValueError, "")

  pow(2.0, (n - 1).float()).uint64()

proc total*: uint64 =
  for i in 1..64:
    result += i.onSquare()
